spec.bs

```diff
@@ -0,0 +1,393 @@
+
+<h3 dfn-type=dfn>Signal base value</h3>
+A signal base value is one of the following:
+<dl dfn-for="signal base value">
+: "<dfn export><code>winning-bid</code></dfn>"
+:: The bid value of the winning bid.
+: "<dfn export><code>highest-scoring-other-bid</code></dfn>"
+:: The bid value of the highest scoring bid that did not win.
+: "<dfn export><code>script-run-time</code></dfn>"
+:: The running time of the script in ms(?).
+: "<dfn export><code>signals-fetch-time</code></dfn>"
+:: The time it took for the signals fetch to complete in ms(?)
+: "<dfn export><code>bid-reject-reason</code></dfn>"
+:: The reason a bid was rejected.
+
+</dl>
+
+Issue: Remove exports when these definitions are used.
+
+Issue: Make sure these definitions match "determine the numeric value" algorithm
+
+Issue: New enum needed for bid reject reasons.
+
+Storage {#storage}
+==================
+
+A user agent holds an <dfn>aggregatable report cache</dfn>, which is an [=ordered set=] of [=aggregatable reports=].
+
+Vendor-specific values {#vendor-specific-values}
+================================================
+
+Issue: Should we use the term "implementation-defined" instead?
+
+<dfn>Max stored reports</dfn> is a positive integer that controls how many unsent reports can be queued waiting.
+
+Issue: More
+
+Algorithms {#algorithms}
+====================
+
+
+<!-- A {{WorkletGlobalScope}} |scope|'s <dfn algorithm for="PrivateAggregation">api identifier</dfn> is the result of:
+1. If |scope|'s [=worklet global scope type=] is {{FledgeWorkletGlobalScope}}, then return "<code>[=worklet type identifier/fledge=]</code>".
+1. [=Assert=]: |scope|'s [=worklet global scope type=] is {{SharedStorageWorkletGlobalScope}}.
+1. Return "<code>[=worklet type identifier/shared-storage=]</code>". -->
+
+To <dfn algorithm for="PrivateAggregation">obtain a reporting endpoint</dfn> given an [=Origin=] |reportingOrigin| and [=worklet type identifier=] |api|:
+1. [=Assert=]: |reportingOrigin| is a [=potentially trustworthy origin=].
+1. Let |path| be the [=string/concatenation=] of «"<code>.well-known/private-aggregation/report-</code>", |api|».
+1. Let |base| be the result on running the [=URL parser=] on the [=serialization of an origin|serialization=] of |reportingOrigin|.
+1. Return the result of running the [=URL parser=] on |path| with |base|.
+
+To <dfn for="PrivateAggregation">obtain an aggregatable report</dfn> given an [=Origin=] |reportingOrigin|, a [=worklet type identifier=] |api| and a [=ordered set=] of {{PAHistogramContribution}} |contributions|:
+1. [=Assert=]: |reportingOrigin| is a [=potentially trustworthy origin=].
+1. Let |reportTime| be the result of running [=PrivateAggregation/obtain a report delivery time=].
+1. Let |report| be a new [=aggregatable report=] with the items:
+    : [=aggregatable report/reporting endpoint=]
+    :: The result of [=PrivateAggregation/obtaining a reporting endpoint=] given |reportingOrigin| and |api|.
```

Comment:
Can we store reporting origin here and construct the URL in the report delivery algorithm?